### PR TITLE
linter fixes

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

**Comment by rfomin** [comment]
Report with warnings: https://github.com/fabiangreffrath/woof/actions/runs/9313491371/attempts/1#summary-25635970365

**Comment by fabiangreffrath** [comment]
> Report with warnings: https://github.com/fabiangreffrath/woof/actions/runs/9313491371/attempts/1#summary-25635970365

That's cool!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -178,7 +178,24 @@ jobs:
...
+          version: 18
+          step-summary: true
+          files-changed-only: false
+          style: ''  # disable clang-format checks.
+          tidy-checks: >-
+            -*
+            ,clang-analyzer-*
+            ,-clang-analyzer-cplusplus*
+            ,-clang-analyzer-security*
+            ,-clang-analyzer-valist*
```
Comment (rfomin) [inline .github/workflows/main.yml:195]: Iy seems `valist` checks are buggy: https://github.com/llvm/llvm-project/issues/40656

File: src/mn_menu.c [hunk 2]
```diff
@@ -1522,8 +1522,6 @@ static void M_EndGame(int choice)
 
 static void M_ChangeMessages(int choice)
 {
-    // warning: unused parameter `int choice'
-    choice = 0;
```
Comment (fabiangreffrath) [inline src/mn_menu.c:1526]: This was once added to fix another warning. I think we could just invalidate the variable by calling `(void)choice;`.

Comment (rfomin) [reply src/mn_menu.c:1526]: Is this a "pedantic" warning? I think both warnings are unnecessary, we can disable this clang-tidy check (although it did find some dead code).

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -178,7 +178,24 @@ jobs:
...
+          database: 'build'
+          ignore: 'miniz|spng|win32|toolsrc|src/i_pcsound.c'
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:197]: Why do we exclude `src/i_pcsound.c`?

Comment (rfomin) [reply .github/workflows/main.yml:197]: i_pcsound.c only works if `HAVE_AL_BUFFER_CALLBACK` is defined (ubuntu-24.04 doesn't have it). Although clang-tidy supports a compiler database, it lacks any file filtering, so we have to filter it manually.

**Comment by rfomin** [comment]
I've fixed most of the warnings, the rest are not so trivial or related to padding (I don't think it's worth rearranging those structs). So I removed the workflow failing, maybe someone will fix the warnings one day.

See the full report in the workflow summary: https://github.com/fabiangreffrath/woof/actions/runs/9606782914/attempts/1#summary-26496852091

File: .github/workflows/main.yml (hunk 1 above)
Comment (fabiangreffrath) [reply .github/workflows/main.yml:181]: So, are you going disable them?

**Review by fabiangreffrath — APPROVED** [review]
Fine with me, thanks!

File: .github/workflows/main.yml (hunk 1 above)
Comment (rfomin) [reply .github/workflows/main.yml:181]: I already disabled them. `-clang-analyzer-*` means we enable all `analyzer` checks, `-clang-analizer-valist*` means we disable all `valist` checks.

Comment (fabiangreffrath) [reply .github/workflows/main.yml:181]: Ah, I overlooked the `-` prefix to the command.

**Pull request merged**